packages/@react-aria/interactions/src/usePress.ts

```diff
@@ -774,8 +773,35 @@ function shouldPreventDefault(target: Element) {
   return !(target instanceof HTMLElement) || !target.draggable;
 }
 
-function shouldPreventDefaultKeyboard(target: Element) {
-  return !((target.tagName === 'INPUT' || target.tagName === 'BUTTON') && (target as HTMLButtonElement | HTMLInputElement).type === 'submit');
+function shouldPreventDefaultKeyboard(target: Element, key: string) {
+  if (target instanceof HTMLInputElement) {
+    return !isValidInputKey(target, key);
+  }
+
+  if (target instanceof HTMLButtonElement) {
+    return target.type !== 'submit';
+  }
+
+  return true;
+}
+
+const nonTextInputTypes = new Set([
+  'checkbox',
+  'radio',
+  'range',
+  'color',
+  'file',
+  'image',
+  'button',
+  'submit',
+  'reset'
+]);
```

Comment:
How was this list generated? I assume by going through a list of input types and adding any that wasn't a text input?

Reply:
yeah https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input